ResourceStatusSystem/Client/ResourceStatus.py

```diff
@@ -124,72 +152,70 @@ def __updateRssCache( self ):
 
 ################################################################################
 
-  def __getRSSElementStatus( self, elementName, elementType, statusTypes ):
-    """
-    Gets from the cache or the RSS the Elements status. The cache is a
-    copy of the DB table. If it is not on the cache, most likely is not going
-    to be on the DB.
-
-    There is one exception: item just added to the CS, e.g. new Element.
-    The period between it is added to the DB and the changes are propagated
-    to the cache will be inconsistent, but not dangerous. Just wait <cacheLifeTime>
-    minutes.
-    """
+  def __getRSSElementStatus( self, elementName, elementType, statusType ):
+    """ Gets from the cache or the RSS the Elements status. The cache is a
+        copy of the DB table. If it is not on the cache, most likely is not going
+        to be on the DB.
+
+        There is one exception: item just added to the CS, e.g. new Element.
+        The period between it is added to the DB and the changes are propagated
+        to the cache will be inconsistent, but not dangerous. Just wait <cacheLifeTime>
+        minutes.
+
+  :param elementName: name of the element
+  :type elementName: str
+  :param elementType: type of the element (StorageElement, CE, FTS, Catalog)
+  :type elementType: str
+  :param statusType: type of the status (meaningful only when elementType==StorageElement, otherwise it is 'all' or ['all'])
+  :type statusType: str, list
+  """
 
-    cacheMatch = self.rssCache.match( elementName, elementType, statusTypes )
+    cacheMatch = self.rssCache.match( elementName, elementType, statusType )
 
     self.log.debug( '__getRSSElementStatus' )
     self.log.debug( cacheMatch )
 
     return cacheMatch
 
   def __getCSElementStatus( self, elementName, elementType, statusType, default ):
-    """
-    Gets from the CS the Element status
+    """ Gets from the CS the Element status
+
+    :param elementName: name of the element
+    :type elementName: str
+    :param elementType: type of the element (StorageElement, CE, FTS, Catalog)
+    :type elementType: str
+    :param statusType: type of the status (meaningful only when elementType==StorageElement)
+    :type statusType: str, list
+    :param default: defult value
+    :type default: None, str
     """
 
-    if elementType in ('ComputingElement', 'FTS'):
+    # DIRAC doesn't store the status of CEs nor FTS in the CS, so here we can just return 'Active'
+    if elementType in ('CE', 'FTS'):
       return S_OK( { elementName: { (elementType, 'all'): 'Active'} } )
 
-    cs_path = "/Resources/" + elementType
+    # If we are here it is because elementType is either 'StorageElement' or 'Catalog'
+    if elementType == 'StorageElement':
+      cs_path = "/Resources/StorageElements"
+    elif elementType == 'Catalog':
+      cs_path = "/Resources/FileCatalogs"
+      statusType = ['Status']
```

Comment:
No it's not needed for StorageElements. And the fact that it's needed for FileCatalogs is because we have a spaghetti CS.